src/wasm2asm.h

```diff
@@ -1320,6 +1167,18 @@ Ref Wasm2AsmBuilder::processFunctionBody(Function* func, IString result) {
     Ref visitConst(Const* curr) {
       switch (curr->type) {
         case i32: return ValueBuilder::makeInt(curr->value.geti32());
+        // An i64 argument translates to two actual arguments to asm.js
+        // functions, so we do a bit of a hack here to get our one `Ref` to look
+        // like two function arguments.
+        case i64: {
+          unsigned lo = (unsigned) curr->value.geti64();
+          unsigned hi = (unsigned) (curr->value.geti64() >> 32);
+          std::ostringstream out;
+          out << lo << "," << hi;
+          std::string os = out.str();
+          IString name(os.c_str(), false);
+          return ValueBuilder::makeName(name);
```

Comment:
i don't understand what this does. it returns a `name` containing `low,high`? how does that work later?

Reply:
Yeah this is sort of a hack and abuse of `makeName`. The `makeName` emits the raw string into the JS without much other ado, so this needs to somehow emit two arguments in one `Ref` (to avoid a larger refactoring) which ends up with a hack like this. If you'd prefer I can try to figure out a way that doesn't require the hack